modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -219,6 +224,9 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
 
                 jg.writeEndObject();
             }
+        } catch (DocumentNotFoundException e) {
+            // If the DocumentRoute is not found (deleted for example), the TaskWriter fails.
+            log.error(e, e);
```

Comment:
If a sysadmin can't do much about it and the application can keep functioning, I'd prefer a WARN.

Also where is the exception being thrown exactly? Could you add a full stack trace in the NXP ticket please. I'm asking because the try/catch has a very large scope, which is bad if we can have a smaller one.
From a quick test it seems to appear in `nodeAccessRunner.runUnrestricted()` so the try/catch should be around that, and you should `log.warn("Failed to get workflow instance", e)`

Reply:
Making it a warn and updating the ticket!

Yes indeed, I just catched there because there was already a try clause and because I was certain about the context this is called in (by context I mean while writing a task to json). The `UnrestrictedSessionRunner.runUnrestricted()` is called in so many places I didn't want to catch in it.

I could also make a sub try catch to make sure I only around the `nodeAccessRunner.runUnrestricted();`call in `TaskWriter.writeEntityBody()` line 116. At first I wanted to avoid doing a sub try catch.

Reply:
Yes I mean around the call to `nodeAccessRunner.runUnrestricted` at line 116. It helps the reader to know where the exception can come from.

Reply:
Done!

Reply:
After moving the catch, now the faulty task contains more info in the response. I have updated the jira ticket with a response containing a corrupted task first and then a wellformed one. Also Kevin has identified some issues we'd like to have your opinion about: https://github.com/nuxeo/nuxeo/pull/4008#discussion_r420618445